tests/functional/test_install.py

```diff
@@ -1527,3 +1528,99 @@ def test_target_install_ignores_distutils_config_install_prefix(script):
         (target - script.base_path) in result.files_created and
         (prefix - script.base_path) not in result.files_created
     ), str(result)
+
+
+@pytest.mark.skipif("sys.platform != 'win32'")
+@pytest.mark.parametrize('pip_name', [
+    'pip',
+    'pip{}'.format(sys.version_info[0]),
+    'pip{}.{}'.format(*sys.version_info[:2]),
+    'pip.exe',
+    'pip{}.exe'.format(sys.version_info[0]),
+    'pip{}.{}.exe'.format(*sys.version_info[:2])
+])
+def test_protect_pip_from_modification_on_windows(script, pip_name):
+    """
+    Test ``pip install --upgrade pip`` is raised an error on Windows.
+    """
+    command = [pip_name, 'install', '--upgrade', 'pip']
+    result = script.run(*command, expect_error=True)
+    assert result.returncode != 0
+    new_command = [sys.executable, '-m', 'pip'] + command[1:]
+    assert 'To modify pip, please run the following command:\n{}'.format(
+        ' '.join(new_command)) in result.stderr, str(result)
+
+
+@pytest.mark.skipif("sys.platform != 'win32'")
+def test_protect_pip_from_modification_via_deps_on_windows(script, with_wheel):
+    """
+    Test ``pip install pkga`` is raised an error on Windows
+    if `pkga` implicitly tries to upgrade pip.
+    """
+    # Make a wheel for pkga which requires pip
```

Comment:
We have an existing helper for this: `create_basic_wheel_for_package`, which doesn't depend on setuptools or creating a subprocess, for creating the wheel.

```py
from tests.lib import create_basic_wheel_for_package

pkg_A_path = create_basic_wheel_for_package(
    script,
    "A", "0.0.1",
    depends=["pip != {}".format(pip.__version__)],
)
```

Reply:
Great, i was thinking about some kind of fixture, but this is exactly what i need!